runtime/lua/vim/lsp/util.lua

```diff
@@ -522,7 +522,7 @@ function M.apply_text_edits(text_edits, bufnr, offset_encoding)
   if is_cursor_fixed then
     local is_valid_cursor = true
     is_valid_cursor = is_valid_cursor and cursor.row < max
-    is_valid_cursor = is_valid_cursor and cursor.col <= #(get_line(bufnr, max - 1) or '')
+    is_valid_cursor = is_valid_cursor and cursor.col <= #(get_line(bufnr, cursor.row) or '')
```

Comment:
I tried to add a test case for this fix, but decided the problem was too simple to be worth adding as a separate test case.

Reply:
It would still be nice to have a test case to catch regressions in the future - in case the method gets refactored.

Reply:
OK. I'll add it. Thank you for advice.